src/lib.rs

```diff
@@ -224,6 +224,20 @@ pub fn consumer(
                     panic!("Invalid content type");
                 }
             },
+            ItemFormat::JACOCO_XML => match work_item.item {
```

Comment:
I can't think of a way to avoid the if, it has to be somewhere. I would share the code anyway here, even if then we have to have an if.

Reply:
the producer only uses ItemType::Content and not ItemType::Path.
That means that I/O stuff for jacoco files is done one the producer thread so if it isn't a problem we can just put a panic in ItemType:Path in the consumer.

Reply:
@calixteman That's right. Same thing with `ItemFormat::INFO`.

Reply:
Yep exactly (about INFO files).
@marco-c should we keep the file reading in the producer ?
FYI I let it like that for info files because they're not so much and because that avoided to make a distinction between info files coming from zip (so a buffer) and ones coming from dir (so a file).

Reply:
The jacoco xml reports are about 8M, probably won't grow much more than that.

Reply:
ok so nothing ;)
So maybe you should try to remove ItemType::Path